File: src/main/java/io/braendli/importer/ImportForm.java
package io.braendli.importer;

import javafx.application.Application;
import javafx.fxml.FXML;
import javafx.fxml.FXMLLoader;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.scene.control.Button;
import javafx.scene.control.CheckBox;
import javafx.scene.control.TextField;
import javafx.scene.image.Image;
import javafx.scene.layout.GridPane;
import javafx.stage.FileChooser;
import javafx.stage.FileChooser.ExtensionFilter;
import javafx.stage.Stage;
import org.controlsfx.control.NotificationPane;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.File;
import java.io.IOException;

public class ImportForm extends Application {
    private static final Logger LOG = LoggerFactory.getLogger(ImportForm.class);

    private Stage stage;
    private NotificationPane notification;
    @FXML
    private GridPane mainPane;
    @FXML
    private CheckBox deleteOldDataBox;
    @FXML
    private Button importButton;
    @FXML
    private TextField excelField;
    @FXML
    private TextField databaseField;

    @FXML
    public void handleChooseExcel() {
        FileChooser fileChooser = new FileChooser();
        fileChooser.setTitle("Excel Datei wählen");
        fileChooser.getExtensionFilters().add(new ExtensionFilter("Excel Dateien", "*.xlsx"));
        updateTextField(excelField, fileChooser.showOpenDialog(stage));
    }

    @FXML
    public void handleChooseDatabase() {
        FileChooser fileChooser = new FileChooser();
        fileChooser.setTitle("Datenbank wählen");
        fileChooser.getExtensionFilters().add(new ExtensionFilter("Datenbanken", "*.FDB"));
        updateTextField(databaseField, fileChooser.showOpenDialog(stage));
    }

    private void updateTextField(TextField field, File file) {
        if (file != null && file.exists()) {
            field.setText(file.getAbsolutePath());
        }
    }

    @FXML
    public void initialize() {
        updateTextField(excelField, new File(System.getProperty("user.home"), "Desktop\\users.xlsx"));
        updateTextField(excelField, new File("users.xlsx"));
        updateTextField(databaseField, new File("C:\\Program Files\\SafeScan\\TA\\TADATA.FDB"));
        importButton.disableProperty().bind(excelField.textProperty().isEmpty().or(databaseField.textProperty().isEmpty()));
    }

    public static void main(String[] args) throws ClassNotFoundException {
        launch(args);
    }

    @Override
    public void start(Stage stage) throws IOException {
        FXMLLoader loader = new FXMLLoader(getClass().getResource("/import_form.fxml"));
        Parent root = loader.load();
        NotificationPane notification = new NotificationPane(root);
        ((ImportForm) loader.getController()).setUp(stage, notification);
        Scene scene = new Scene(notification, 500, -1);
        stage.getIcons().add(new Image(getClass().getResourceAsStream("/icons/app.png")));
        stage.setTitle("Excel TA Import");
        stage.setScene(scene);
        stage.show();
    }

    private void setUp(Stage stage, NotificationPane notification) {
        this.stage = stage;
        this.notification = notification;
    }

    public void importData() {
        LOG.info("Starting import");
        try {
            Importer.importToDatabase(
                    deleteOldDataBox.isSelected(),
                    new File(excelField.getText()),
                    new File(databaseField.getText())
            );
            notification.setText("Import erfolgreich abgeschlossen");
            LOG.info("Import successful");
        } catch (Exception e) {
            notification.setText("Der Import ist fehlgeschlagen");
            LOG.error("Import failed", e);
        } finally {
            notification.show();
        }
    }
}

File: src/main/java/io/braendli/importer/ImportForm.kt
package io.braendli.importer

import javafx.application.Application
import javafx.fxml.FXML
import javafx.fxml.FXMLLoader
import javafx.scene.Parent
import javafx.scene.Scene
import javafx.scene.control.Button
import javafx.scene.control.CheckBox
import javafx.scene.control.TextField
import javafx.scene.image.Image
import javafx.stage.FileChooser
import javafx.stage.FileChooser.ExtensionFilter
import javafx.stage.Stage
import org.controlsfx.control.NotificationPane
import org.slf4j.LoggerFactory
import java.io.File
import java.io.IOException

class ImportForm : Application() {
    private lateinit var stage: Stage
    private lateinit var notification: NotificationPane
    @FXML
    private lateinit var deleteOldDataBox: CheckBox
    @FXML
    private lateinit var importButton: Button
    @FXML
    private lateinit var excelField: TextField
    @FXML
    private lateinit var databaseField: TextField

    companion object {
        private val LOG = LoggerFactory.getLogger(ImportForm::class.java)

        @JvmStatic
        fun main(args: Array<String>) {
            Application.launch(ImportForm::class.java)
        }
    }

    @FXML
    fun handleChooseExcel() {
        val fileChooser = FileChooser()
        fileChooser.title = "Excel Datei wählen"
        fileChooser.extensionFilters.add(ExtensionFilter("Excel Dateien", "*.xlsx"))
        updateTextField(excelField, fileChooser.showOpenDialog(stage))
    }

    @FXML
    fun handleChooseDatabase() {
        val fileChooser = FileChooser()
        fileChooser.title = "Datenbank wählen"
        fileChooser.extensionFilters.add(ExtensionFilter("Datenbanken", "*.FDB"))
        updateTextField(databaseField, fileChooser.showOpenDialog(stage))
    }

    private fun updateTextField(field: TextField, file: File?) {
        if (file != null && file.exists()) {
            field.text = file.absolutePath
        }
    }

    @FXML
    fun initialize() {
        updateTextField(excelField, File(System.getProperty("user.home"), "Desktop\\users.xlsx"))
        updateTextField(excelField, File("users.xlsx"))
        updateTextField(databaseField, File("C:\\Program Files\\SafeScan\\TA\\TADATA.FDB"))
        importButton.disableProperty().bind(excelField.textProperty().isEmpty.or(databaseField.textProperty().isEmpty))
    }

    @Throws(IOException::class)
    override fun start(stage: Stage) {
        val loader = FXMLLoader(javaClass.getResource("/import_form.fxml"))
        val root = loader.load<Parent>()
        val notification = NotificationPane(root)
        (loader.getController<Any>() as ImportForm).setUp(stage, notification)
        val scene = Scene(notification, 500.0, -1.0)
        stage.icons.add(Image(javaClass.getResourceAsStream("/icons/app.png")))
        stage.title = "Excel TA Import"
        stage.scene = scene
        stage.show()
    }

    private fun setUp(stage: Stage, notification: NotificationPane) {
        this.stage = stage
        this.notification = notification
    }

    fun importData() {
        LOG.info("Starting import")
        try {
            Importer.importToDatabase(
                    deleteOldDataBox.isSelected,
                    File(excelField.text),
                    File(databaseField.text)
            )
            notification.text = "Import erfolgreich abgeschlossen"
            LOG.info("Import successful")
        } catch (e: Exception) {
            notification.text = "Der Import ist fehlgeschlagen"
            LOG.error("Import failed", e)
        } finally {
            notification.show()
        }
    }
}
